sketch_main.cpp

```diff
@@ -200,7 +202,7 @@ int main(int argc, char *argv[]) {
     } else if (FLAGS_sketch_method.starts_with("Tensor")) {
         Tensor<uint64_t> tensor_sketch(kmer_word_size, FLAGS_embed_dim, FLAGS_tuple_length);
         TensorSlide<uint64_t> tensor_slide(kmer_word_size, FLAGS_embed_dim, FLAGS_tuple_length,
-                                           FLAGS_window_size, FLAGS_stride);
+                                           FLAGS_window_size, FLAGS_stride, FLAGS_seq_len);
```

Comment:
Sequences may have different lengths so passing this on to TensorSlide can't be right. Since this is a public interface  we really have to be careful what we expose to potential users.